### DISCUSSION TITLE
Network failure

### DISCUSSION THREAD

**Discussion by Davey1970** [xdream-biking/trixter-xdream-bike#104] [Q&A]
I changed the button battery on the PC because it was keep alerting. After I changed it I keep getting a network failure when I try to add a user. It all works OK in guest mode but I cannot add a user. Any ideas how to fix this.

**Comment by drmason789** [comment]
Hello,

Are you able to determine which version of the software you are using? See #18.

**Reply by Davey1970** [reply]
I think it’s V1

**Reply by Davey1970** [reply]
Sorry my mistake it says : v3.02

**Reply by Davey1970** [reply]
So I assume it’s version 3

**Reply by Davey1970** [reply]
After I enter the user details it comes up with an error and states’ due to a network failure we are unable to save your user file. Please try again later’

**Comment by Davey1970** [comment]
It’s V3.02


After I enter the user details it comes up with an error and states’ due to
a network failure we are unable to save your user file. Please try again
later’

**Comment by drmason789** [comment]
Version 3 of the software saves user data in the Postgres database server instance.

Are you able to go to Control Panel | Administrative Tools | Services and check if the PostgreSQL service is running?

![image](https://github.com/drmason789/trixter-xdream-bike/assets/29954900/03571547-6a49-49a7-acd3-7f48ab9d5dac)

It might look a bit different as I'm running that in Windows 10 on a different computer with a slightly later version of Postgresql than the one on the Trixter PC.

**Comment by drmason789** [comment]
Also, you haven't explicitly stated that adding users ever worked for you. 

The software also looks at a server called "trixter01" for the database. If you look in the computer's hosts file, you may see this is assigned an IP address for a server that doesn't exist on your network. In that case, the solution may be to delete the entry from the hosts file and get the database server up and running on your PC.

See #63 for details of the hosts file.

**Reply by Davey1970** [reply]
Yes it did work before. I cannot see PostgreSQL in the services file

**Reply by drmason789** [reply]
Have you checked the hosts file (#63) ?

**Reply by Davey1970** [reply]
I can see the HOST file. Do I need to change it to Trixter01

**Reply by Davey1970** [reply]
In the etc file there is: lmhosts, networks, protocol, services and hosts

**Reply by drmason789** [reply]
It's the "hosts" one - it maps names to I addresses. If you don't have trixter01 in there, X-Dream will look at the local computer AFAIK, so if it's not there you shouldn't need to change it.

Do you have this folder?

`C:\Program Files\PostgreSQL`

**Reply by drmason789** [reply]
Has anything else changed in the environment since you changed the button cell? Notably, another X-Bike PC or some other server that's been removed from the network? Because even if there's no "trixter01" in the hosts file, there could have been a trixter01 PC on your network that was receiving these requests.

**Reply by Davey1970** [reply]
I can see there is Trixter01 to Trixter15. The Trixter01 is at address 192.168.32.1

**Reply by Davey1970** [reply]
All I did was change the button cell

**Reply by Davey1970** [reply]
Nothing else has changed

**Reply by Davey1970** [reply]
I can see the file C:\Program Files\PostgreSQL
In there is 9.1

**Reply by Davey1970** [reply]
In 9.1 there is data, lib, stackbuilder, doc, pgadminIII, symbols, include and share

**Reply by drmason789** [reply]
So it's installed, but you didn't see it in the Services list. Are you able to run Start | PostGreSQL 9.1 | pgAdmin III ? I'm guessing at these names based on a 9.5 installation in Windows 10 - I don't want to extract the Trixter PC from mothballs just yet.

If you are prompted for username/password use "postgres".

![image](https://github.com/drmason789/trixter-xdream-bike/assets/29954900/6470484e-8646-4400-9e14-412d1dca9929)

**Reply by drmason789** [reply]
Also, is the PC plugged into a network now, when it wasn't before, or vice versa? Or perhaps a router it's connected to altered in some way?

**Reply by Davey1970** [reply]
When I run it I don’t get username/password request. It goes to the Trixter screen and on the left I have recycle bin, XDream test and shortcut for Trixter xdream biking. I’m the pgadminIIi is only two files docs and i18n they have a few different countries listed. If I click on en_US there are two files hints and pgadmin3 ( compiled HTML Help file)

**Reply by Davey1970** [reply]
I cannot see the object browser

**Reply by Davey1970** [reply]
No it’s never been on a network or router before

**Reply by Davey1970** [reply]
If I go into the network connection it’s just a local area connection with a Red Cross between the two screens

**Reply by Davey1970** [reply]
I think the Trixter screen and files on the left are just underneath the new files so ignore that

**Reply by drmason789** [reply]
Are the date and time set correctly?

**Reply by Davey1970** [reply]
Yes the date and time are correct

**Reply by drmason789** [reply]
There are also some config files in C:\Program Files\Trixter\Live\Scripts matching the pattern MainXD*.txt.

I believe the one to look in for v3 is MainXD30aN.txt. In this file there are some lines specifying paths for user data. I have not actually experimented with these at all, and I don't know if V3 does anything with them.  I think these are actually for version 2 of the software.

```
UserNetworkPath		"\\trixter01\Users\Public\Documents\allusers"		//Place where user profiles are stored (can be networked)
UserNetworkBackupPath	"\\trixter02\Users\Public\Documents\backup"		//Place where user profiles are stored (can be networked)
XPUserNetworkPath	"\\trixter01\SharedDocs\allusers"			//Place where user profiles are stored (can be networked)
XPUserNetworkBackupPath	"\\trixter02\SharedDocs\backup"				//Place where user profiles are stored (can be networked)
UserPathRemote		1			//Sets game to use Networked paths (taken from above)
```

If you look in the UserNetworkPath, you might see an XML file with your login, and if its modified date is from last time you successfully used it, this could mean the software is configured to use the filesystem rather than the PostgreSQL database.

**Reply by Davey1970** [reply]
I found usernetworkpath and next to it is “x:\allusers” then \\place where user profiles are stored (can be networked) there doesn’t seem to be any date on it

**Reply by drmason789** [reply]
> When I run it I don’t get username/password request. It goes to the Trixter screen and on the left I have recycle bin, XDream test and shortcut for Trixter xdream biking. I’m the pgadminIIi is only two files docs and i18n they have a few different countries listed. If I click on en_US there are two files hints and pgadmin3 ( compiled HTML Help file)

Were you perhaps looking in the filesystem rather than the Start Menu? If you don't see it in the Start menu, you could try opening 
C:\Program Files\PostgreSQL\9.1\bin\pgAdmin3.exe directly.

**Reply by Davey1970** [reply]
I tried to find it by the start menu but It cannot be found. There is no exe file. There is no bin\pgAdmin3.exe 
I can only get up to 9.1 then the two files docs and i18n.

**Reply by Davey1970** [reply]
There is no bin file

**Reply by drmason789** [reply]
In the filsystem  of my original Trixter PC, docs and i18n are in 
   "C:\Program Files\PostgresSQL\9.1\pgAdmin III"

I'm asking you to look in:
  "C:\Program Files\PostgreSQL\9.1\bin" 
for the pgAdmin3.exe file (and you might not see the .exe extension if file extensions are not being shown).

**Reply by Davey1970** [reply]
I looked for C:\Program Files\PostgreSQL\9.1\bin but there is no file that name. There is no bin folder in 9.1. I tried searching for pgAdmin3.exe but again no file by that name. I may be missing the bin folder with pgAdmin3.exe

**Reply by Davey1970** [reply]
The PC is running windows XP if it makes a difference

**Reply by drmason789** [reply]
I've never seen an XP installation. Have you attempted to look in X:\allusers, or the path specified in the config file for XPUserNetworkPath?
Also, what value do you have for UserPathRemote?

Also, there's a file C:\Program Files\Trixter\Live\Scripts\Service.txt which contains an entry like
`main_txt	"MAINXD30AN.TXT"`

I believe this is where the main*.txt file that it actually uses is specified. Does this actually specify MAINXD30AN.TXT?

**Reply by Davey1970** [reply]
There is no x:\\ drive only c:\\ 
The userpathremote is 1 
Yes MAINXD30AN.TXT is specified in main_txt

**Reply by drmason789** [reply]
> The PC is running windows XP if it makes a difference

I don't know if PostgreSQL 9.1 will even run on Windows XP - my original Trixter PC is running Windows 7. Currently I don't know where your XDream installation has been storing user data. 

If it's never been on a network while you've successfully been logged in as a user, and PostgreSQL isn't working, I expect it must be storing the information in the filesystem somewhere, and that X:\Allusers from the mainXD30an.txt is probably a good place to look. The user data I've seen is in the form of username.xml. I'm not sure where to look in Windows XP.

Perhaps you could open a command window and enter the following:

```
CD /D C:\
DIR username.xml /s /b
```

substituting your actual username for "username" above. Or you could just put * instead of username and see all the XML files on the system. UsageStats.xml is another file to look for.

**Reply by Davey1970** [reply]
It could not find X:\\allusers maybe because I do not have an X drive. I only have a C:\\ drive

**Reply by Davey1970** [reply]
Dir*.xml shows says the volume on drive c is OS
And shows the volume serial number.

**Reply by drmason789** [reply]
You need the /s to make it traverse all the subdirectories and /b to just deliver the bare essentials.

`dir *.xml /s /b`

**Reply by drmason789** [reply]
> It could not find X:\allusers maybe because I do not have an X drive. I only have a C:\ drive

X:\ could have been a mapped network drive at one point, and also folders can be mapped to drive letters with the SUBST command. However, you'd see it.

**Reply by Davey1970** [reply]
It doesn’t seem to recognise that. I tried /s and /b together and separately.

**Reply by drmason789** [reply]
Can you use the search facility in Windows Explorer?

**Reply by Davey1970** [reply]
I can use the search using another PC. This one is not connected to the internet

**Reply by drmason789** [reply]
Windows Explorer is the Windows XP equivalent of File Explorer in modern Windows.

**Reply by Davey1970** [reply]
Or do you mean use this one but in offline mode

**Reply by drmason789** [reply]
> It doesn’t seem to recognise that. I tried /s and /b together and separately.

Windows XP should support these options: https://www.oreilly.com/library/view/windows-xp-in/0596002491/re311.html

**Reply by Davey1970** [reply]
Yes I can get into it but only in offline mode because it is not connected to the internet or my router

**Reply by drmason789** [reply]
> Yes I can get into it but only in offline mode because it is not connected to the internet or my router

Are you sure you are referring to [Windows Explorer](https://edu.gcfglobal.org/en/windowsxp/working-with-windows-explorer/1/) and not Internet Explorer?

**Reply by Davey1970** [reply]
Sorry I was getting mixed up. Yes I can search using windows explorer

**Reply by drmason789** [reply]
Are you able to find any interesting xml files such as yourusername.xml or UsageStats.xml, most likely under "C:\Documents and Settings" ?

**Reply by Davey1970** [reply]
There is a couple of usage stats. One is in: Trixter backup localusers and the other is in Trixter live localusers

**Reply by drmason789** [reply]
The one in Live would be the one we want. Is there a file in the same folder that has your username with the .xml extension? Note that you may not see the extension if file extensions are hidden in the Windows Explorer options. To show them see [here](https://fileinfo.com/help/windows_show_extensions).

**Reply by Davey1970** [reply]
There is one with my username I used previously as a xml file

**Reply by drmason789** [reply]
Are you able to determine the date on the username.xml file? It could be in the Date column in Windows Explorer (you might have to adjust the view to "Details" to see this) or right-click | Properties | Details ?

If so, is it approximately the same date and time you were logged in as that username?

**Reply by Davey1970** [reply]
It says accessed on 14th May 2023 modified on 15th May 2023 created 27th May 2019

**Reply by Davey1970** [reply]
I think it was working on 15th May then I changed the battery on 15th May which stopped it working

**Reply by Davey1970** [reply]
I mean 14th working

**Reply by drmason789** [reply]
I think we've found where it's been storing its data. You said in your initial question that you couldn't create new users, not that you couldn't log in as an existing user. Can you log in as an existing user?

I'd like to determine if it has successfully created the user and got the error it is calling a "network error" trying to do something else (e.g. trying to back up this localusers folder). Currently I'm assuming it's got the network error trying to access a UNC (\\\\server\fileshare) path from somewhere in the main config file, although if the computer was never on a network while in your possession, I don't know how it would ever have worked.
- Do you see an xml file for the user you tried to create?
- Is it dated the same time you tried to create the user
- if you open the file with notepad, do you recognise the data you entered?

In your MainDX30aN.txt file, you found the UserNetworkPath value, but because you're using XP, X-Dream is possibly looking at the XPUserNetworkPath value instead. Could you show what you have for the 4 paths?

```
UserNetworkPath		"\\trixter01\Users\Public\Documents\allusers"		//Place where user profiles are stored (can be networked)
UserNetworkBackupPath	"\\trixter02\Users\Public\Documents\backup"		//Place where user profiles are stored (can be networked)
XPUserNetworkPath	"\\trixter01\SharedDocs\allusers"			//Place where user profiles are stored (can be networked)
XPUserNetworkBackupPath	"\\trixter02\SharedDocs\backup"				//Place where user profiles are stored (can be networked)
```

**Reply by Davey1970** [reply]
I cannot access any users not even my last one. My previous username still gives a network failure when I try to login in, same as if I try to enter a new username. Only the guest works. 

There are only two paths :
Usernetworkpath - x:\allusers and userpathremote - 1 

I cannot see the others. They are not in main ether

**Reply by Davey1970** [reply]
If I open the xml with my username the details are recognised as the data I entered

**Reply by Davey1970** [reply]
Just to confirm I cannot enter an existing user or a new user. I can enter as a guest

**Reply by drmason789** [reply]
I now have a suggestion for a change of configuration, but I don't know how it was working before.
1. Make a backup copy of the Live\Scripts\mainDX30an.txt somewhere else (eg. the desktop)
2. Copy (not move) the Live\localusers folder to C:\localusers
3. Edit the Live\Scripts\mainDX30an.txt so that the Usernetworkpath refers to C:\localusers instead of X:\Allusers
4. Try running X-Dream and logging in again, and creating a new user.

**Reply by Davey1970** [reply]
Thanks for your help but it still the same. If you have any other ideas I can try again tomorrow.

**Reply by drmason789** [reply]
My next step would be to restore the Live\Scripts\mainDX30an.txt file from your backup and delete the C:\localusers folder. 
Then use a USB drive to get [Process Monitor](https://learn.microsoft.com/en-us/sysinternals/downloads/procmon) onto the system, with which we can monitor what it's trying to do.

I'm going to be busy tomorrow, but later in the week and next weekend should be possible. 

Additionally, you could consider running the software on a newer computer #5 (I've had it running on an ACEPC-T6 stick PC but it was a bit jerky in parts, ran very well on an Core-i3 system circa 2013). The only problem I've not solved yet is that some of the wmv files that are played - the welcome videos and the trophy videos between levels, don't display properly #16 - they possibly do on Nvidia hardware but I've not tried it yet. You may intermittently have to do some admin on the PL2303 driver #8 #88.

**Reply by Davey1970** [reply]
Thanks I’ll give that a go tomorrow. If I use a newer computer is it straight forward to copy the files or is it more complicated

**Reply by drmason789** [reply]
It's more than just copying the files, as detailed in #5 but the most trouble is the PL2303 driver for the bike which requires registry modifications to hold it back to a version that works, and you also need to configure the COM port. Additionally, you're using a slightly earlier version than the one I made these instructions for so I can't be sure they'll work 100%. I am aware of others who have got the software working with the bike on other computers using my instructions.

Also, since you'd be moving from file user data to PostgreSQL user data, you'll lose it. There may be a way to migrate it (e.g. a v2 to v3 update package) but I haven't found it yet.

On the upside, on a modern computer, using a tool like DxWnd, you can have X-Dream running in a window so you can watch streaming video in one window while riding around the X-Dream track in another. Also, the computer isn't dedicated to X-Dream, it becomes an application you start manually like any other.

**Reply by drmason789** [reply]
@Davey1970 I'd like to check if you've made progress on this. Where are we at? Same as where we left last weekend? Or perhaps you've got it all working on a modern PC and have left the original behind?

**Reply by Davey1970** [reply]
Thanks I really appreciated your help. Still can’t get it to work. I will just use the guest instead as that’s still working.

**Reply by drmason789** [reply]
In case you come back to this:

I've done a bit of further investigating. I have spent the full amount of time I am willing to allocate to getting Windows XP installed and working on any of the spare hardware I have, or Hyper-V or VMWare (I've had X-Dream running in VMWare Workstation), with little success.

I've tried to replicate the problem using an installation of 3.50.0.3 (the version I have) in Windows 11, disabling PostgreSQL and the network, but all I get when trying to create a new user is long pauses and reversion to the date of birth screen. Note: please don't estimate the inconvenience to me based on your experience of doing this. I have an X-Dream bike emulator #57  for which the buttons work much better than the real bike's, and X-Dream itself runs in a window using [DxWnd](https://sourceforge.net/projects/dxwnd/).

I've tried monitoring with ProcMon but I can see it is just repeatedly looking for PostgreSQL, and not any XML files in the filesystem. 
Aside from using ProcMon to see what it's trying to do on your PC, you could possibly check that it doesn't have a problem with its network adapter. I know you don't have it on the network, but if it's trying to use it for something, e.g. to find PostgreSQL or look for other bikes, this could get you a network error. Is it remotely possible it's been disabled in the BIOS or Windows, or you've knocked a jumper or plug on the mainboard when changing the button cell?

Perhaps you could make sure that Live\localusers folder is writable by the login that X-Dream runs under. But this is pre-UAC so I'm not sure there's any risk of that.

There's the possibility of updating the OS to Windows 7 or later, which would let you run PostgreSQL. My Trixter PC has a SATA HDD interface and I've been able to replace the HDD with a modern SSD and also a SATA to SD card adapter (the BIOS objected to the removal of the HDD fan though). Even if your system is older, you can get a replacement IDE HDD cheaply these days, or even an IDE to SD card adapter. That way you can keep your original as a backup.

**Reply by Davey1970** [reply]
Thanks again. I will try updating the OS to windows 7 to see if that works.

**Reply by drmason789** [reply]
@Davey1970 FYI Among developments in the last month, I now have documented solutions in these discussions for all problems I know about running X-Dream in Windows 10 or 11 on modern hardware, notable addition being #16.

**Comment by drmason789** [comment]
@Davey1970 I am now in a better position to answer this. 

The version of X-Dream you report does not support storing user data in a PostgreSQL database.

If you use #106 to find the configuration file, open it and look for UserPathRemote

```
UserNetworkPath		"\\trixter01\Users\Public\Documents\allusers"		//Place where user profiles are stored (can be networked)
UserNetworkBackupPath	"\\trixter02\Users\Public\Documents\backup"		//Place where user profiles are stored (can be networked)
XPUserNetworkPath	"\\trixter01\SharedDocs\allusers"			//Place where user profiles are stored (can be networked)
XPUserNetworkBackupPath	"\\trixter02\SharedDocs\backup"				//Place where user profiles are stored (can be networked)
UserPathRemote		1			//Sets game to use Networked paths (taken from above)
```

If UserPathRemote is 1, and it can't access the the path defined by UserNetworkPath, this could be the problem.
